### ISSUE TITLE
Gallery video should streatch to fill its 

### ISSUE THREAD

**Issue by mushon** [RTLcoil/Turbulence#54]
like this: http://prntscr.com/7fmmzq

**Comment by YomesInc** [comment]
Like that http://prntscr.com/7frmd0 ?

Please verify on the web and let me know

**Comment by mushon** [comment]
@RTLcoil yes.
But please note that the YouTube embed didn't work:
http://prntscr.com/7g0yyy

+the frame with the commission work and artist name should link to the commission (feel free to break this into a different issue if you so prefer)

**Comment by YomesInc** [comment]
Not sure I understand your comment about the link to the commission.

**Comment by YomesInc** [comment]
The link was fixed... twice. Check that all your work is still there, because it usually happens when we overwrite each other.

**Comment by mushon** [comment]
the reason the vimeo link was broken is that you added to it `&autoplay=1&cc_load_policy=1` which should which apparently works for YouTube, but not for Vimeo. The link I entered works:
http://prntscr.com/7h4bq4
It seems like the `&cc_load_policy=1` is the part that broke the video link. I'm not sure how to remove it. It might require JS

**Comment by YomesInc** [comment]
Is Vimeo going to be the only player on the site? Because it currently works with YouTube, like you've mentioned.
We can remove this link but will you be ready to place one like this manually in case you want to autoplay a YouTube video (doesn't apply to Vimeo)?

**Comment by mushon** [comment]
Yes, Turbulence works with Vimeo, so let's optimize for vimeo and we can add a special instruction for YouTube

**Comment by YomesInc** [comment]
Got it, could you check?

**Comment by mushon** [comment]
ok, they both load now but how can we make sure they autoplay too?

**Comment by YomesInc** [comment]
We have three videos on this page, which one would you like to autoplay? Should we freeze the slider as well? At what moment should the video start?

**Comment by mushon** [comment]
All videos here should autoplay (otherwise using an image would do just fine) they should play from the beginning and and loop. I will instruct Turbulence not to use videos with sound here (if the sound can be programmatically muted that would be great but I don't think it can)

**Comment by YomesInc** [comment]
So all three videos on a page should autoplay? That might kill some browsers and it also doesn't make sense to me, but if you want me to do it - let me know and I will.

**Comment by YomesInc** [comment]
I also need to know about slider's behavior as mentioned above.

**Comment by mushon** [comment]
They should autoplay only when they are displayed. They don't need to play in the background. I would also like to play with different delay times, could you point me to where in the JS is that set?

**Comment by mushon** [comment]
Let me know if the video only way you can implement it is to freeze the slider until the video finished. In that case I will instruct Turbulence to only use short videos for the sliders.

**Comment by YomesInc** [comment]
Could you rephrase this? "Let me know if the video only way you can implement it is to freeze the slider until the video finished."
From the way you're asking it I'm not sure if you want or don't want it.

**Comment by mushon** [comment]
Sorry, I was trying to acknowledge the option that it could be that the only way to implement the autoplaying videos is to freeze the slider until each video is finished. In that case I will instruct Turbulence to only use short videos.

**Comment by YomesInc** [comment]
The thing is that with Vimeo autoplay becomes tricky. With Youtube it was possible, but with a free account of Vimeo it's not. If you have a paid account - we can try it there, but for now there's no option to autoplay videos in Vimeo's current account.

Regarding the JS - here's the one for the main slider http://prntscr.com/7khdo4 (filename: main.js)
and here's the second slider, the smaller one http://prntscr.com/7khery

**Comment by mushon** [comment]
Turbulence has a 'PLUS' account with Vimeo: https://vimeo.com/user20654804 do you need credentials for it?

+

Here's a little tip for linking to code references:
main slider:
https://github.com/RTLcoil/Turbulence/blob/gh-pages/wp/wp-content/themes/turbulence/js/main.js#L414

second slider:
https://github.com/RTLcoil/Turbulence/blob/gh-pages/wp/wp-content/themes/turbulence/js/main.js#L484

**Comment by YomesInc** [comment]
Good point, thanks.

And yes, let's try it. Please send me the credentials.

**Comment by YomesInc** [comment]
We've added a simple autoplay, when you open a slide with a video and there is a parameter data-autoplay=1 for that slide http://prntscr.com/7o8zq0 (it is set for the third and not set for the fourth) then the video is launched from the beginning. When the slide changes - the video stops.
By default, slider plays with no stop, but when you hover the navigation 
http://prntscr.com/7o8zg5 then the slider stops

**Comment by mushon** [comment]
Cool. and that would work only for Vimeo, right?

**Comment by YomesInc** [comment]
That's right, we've changed it to Vimeo now.

**Comment by mushon** [comment]
ok, closing :)
